pyo3-macros-backend/src/frompyobject.rs

```diff
@@ -149,7 +149,15 @@ impl<'a> Container<'a> {
             (Fields::Unnamed(_), true) => ContainerType::TupleNewtype,
             (Fields::Unnamed(unnamed), false) => match unnamed.unnamed.len() {
                 1 => ContainerType::TupleNewtype,
-                len => ContainerType::Tuple(len),
+                _ => {
+                    let mut fields = Vec::new();
```

Comment:
I think this can simplified using `collect`, i.e.
```rust
let fields = unnamed.unnamed.iter().map(|field| FieldPyO3Attributes::from_attrs(&field.attrs)).collect::<Result<Vec<_>>>()?;
```

Reply:
Agreed, I tried that before and for some reason I failed to get it compiled. I've adjusted it.